### ISSUE TITLE
Don't re-render subtrees who's size don't change on state update

### ISSUE THREAD

**Issue by emilsjolander** [facebook/litho#76]
If we know that the size of a subtree is not changing when we update its state, then we definitely don't need to relayout the whole tree, which should save a lot of work.

We could check for the prefix of the global key of a root and if we don't have any pending state updates for it then just copy everything from the old tree. 

### Code pointers
https://github.com/facebook/litho/blob/master/litho-core/src/main/java/com/facebook/litho/ComponentTree.java

https://github.com/facebook/litho/blob/master/litho-core/src/main/java/com/facebook/litho/Component.java#L298

https://github.com/facebook/litho/blob/master/litho-core/src/main/java/com/facebook/litho/LayoutState.java

Point of contact: @mihaelao

**Comment by appleguy** [comment]
@mihaelao curious if this optimization has been implemented, or is still future work.

I found it was difficult to implement this when using Yoga in the Texture framework, since the constraints used for items in the middle of the tree are not exposed outside of Yoga. However, with the flexbox-based ASLayoutSpec system in Texture, this optimization proved very effective.

In the latter case, the API contract is simply that parents must inform their children if _constraints_ change. Until then, children can 1. re-measure with the same constraints, and 2. if the resulting size is the same, there is no need to invalidate the tree above.
